File: app/javascript/types/Post.ts
type PostType = {

    id: number,
    created_at: string,
    updated_at: string,
    user_id: number,
    title: string,
    content: string,
    comments: CommentType[]
}

type CommentType = {
    id: number,
    created_at: string
    updated_at: string,
    post_id: number,
    user_id: number,
    content: string,
    user: {
        username: string
    }
}

export default PostType;

File: app/models/post.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: posts
#
#  id         :bigint           not null, primary key
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  user_id    :integer          not null
#  title      :string(255)      not null
#  content    :text(65535)      not null
#
class Post < ApplicationRecord
  belongs_to :user

  validates :title, presence: true, length: { maximum: 200 }
  validates :content, presence: true, length: { maximum: 2000 }
  validates :user_id, presence: true

  has_many :comments, dependent: :destroy
end
